packages/api/src/services/study.service.ts

```diff
@@ -119,11 +119,43 @@ export const putTask = async (
     return [201, populated];
   } else {
     const task = await CustomizedBattery.create({
-      taskData,
       _id: taskId,
+      ...taskData,
     });
     await study.updateOne({ $push: { batteries: task._id } });
     const populated = await task.populate<{ battery: IBattery }>("battery");
     return [200, populated];
   }
 };
+
+export const getVariant = async (serverCode: string): APIResponse<object> => {
+  if (!serverCode) {
+    throw new HttpError(400, "Missing serverCode in query parameters");
+  }
+
+  const study = await Study.findOne({
```

Comment:
This works, but I wonder if we'd be able to query directly on variants by server code rather than go through the studies themselves.

Reply:
I don't think we can because right now variants are only contained inside the Study model